core/store/src/trie/shard_tries.rs

```diff
@@ -751,7 +751,7 @@ impl WrappedTrieChanges {
         WrappedTrieChanges { tries, shard_uid, trie_changes, state_changes, block_hash }
     }
 
-    pub fn state_changes(&self) -> &[RawStateChangesWithTrieKey] {
+    pub fn state_changes(&self) -> &Vec<RawStateChangesWithTrieKey> {
```

Comment:
that seems weird, please elaborate why do you need this change

Reply:
I'm passing state_changes into a function as `state_changes: &Vec<RawStateChangesWithTrieKey>,` and this was causing a type mismatch.

Reply:
agreed to change `save_flat_state_changes` api